coffee-chats/src/main/webapp/src/components/FindOptimalTimeDialog.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useState} from "react";
+import {Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Grid} from "@material-ui/core";
+import {MultiDatePicker} from "./MultiDatePicker";
+import {TimePicker} from "@material-ui/pickers";
+import {MaterialUiPickersDate} from "@material-ui/pickers/typings/date";
+import {getHours, getMinutes, setHours, setMinutes, startOfDay} from "date-fns";
+import {postData} from "../util/fetch";
+import {Group} from "../entity/Group";
+
+interface FindOptimalTimeDialogProps {
+  open: boolean;
+  setOpen: (value: boolean) => void;
+  setDate: (value: Date) => void;
+  group: Group;
+  duration: number; // in minutes
+}
+
+export function FindOptimalTimeDialog({open, setOpen, setDate, group, duration}: FindOptimalTimeDialogProps) {
+  const [dates, setDates] = useState([new Date()]);
+
+  const [timeStart, setTimeStart] = useState<MaterialUiPickersDate>( // from 9:00
+      setHours(setMinutes(new Date(), 0), 9)
```

Comment:
Would creating new Date not also set seconds and milliseconds to some value? This may cause weird calendar behaviour if randomly it appears to be 9:00am but is in fact 9:00.40 am causing there to be a "free" slot of 40 seconds at 9am.

Reply:
That is true, but that should not get to the server, this is just for the material ui time picker.

When I send the data to the server I align it to the day boundary and then add hours and minutes (so that the seconds will be zero)

Reply:
I see, could you add a test to ensure that timezones are handled correctly? It isn't immediately clear if setting the start of day works when say setting from a far west TimeZone, but I think it should if the dates internally stay in UTC but display in local time.